lib/fizzy/execute.cpp

```diff
@@ -213,8 +224,56 @@ execution_result execute(Instance& instance, FuncIdx function, std::vector<uint6
             goto end;
         case Instr::nop:
             break;
+        case Instr::block:
+        {
+            const auto arity = read<uint8_t>(immediates);
+
+            if (arity != 0)
+            {
+                throw std::runtime_error(
+                    "block instruction with non-empty result type not implemented yet");
+            }
+
+            const auto target_pc = read<uint32_t>(immediates);
+            const auto target_imm = read<uint32_t>(immediates);
+            LabelContext label{code.instructions.data() + target_pc,
+                code.immediates.data() + target_imm, arity, stack.size()};
+            labels.emplace_back(label);
+            break;
+        }
+        case Instr::loop:
+        {
+            LabelContext label{pc - 1, immediates, 0, stack.size()};  // Target this instruction.
+            labels.push_back(label);
+            break;
+        }
         case Instr::end:
-            goto end;
+        {
+            if (!labels.empty())
+                labels.pop_back();
+            else
+                goto end;
+            break;
+        }
+        case Instr::br:
+        case Instr::br_if:
+        {
+            const auto labelidx = read<uint32_t>(immediates);
```

Comment:
In some parts we have used underscore before `idx` (e.g. `label_idx`), but I don't care which one.